nixos/modules/services/monitoring/prometheus/exporters/redfish.nix

```
@@ -0,0 +1,79 @@
+{
+  config,
+  lib,
+  pkgs,
+  options,
+  ...
+}:
+
+let
+  cfg = config.services.prometheus.exporters.redfish;
+  inherit (lib)
+    mkOption
+    types
+    mkPackageOption
+    optionalString
+    concatStringsSep
+    optionalAttrs
+    ;
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "redfish-config.yml" cfg.settings;
+in
+{
+  port = 9610;
+  extraOpts = {
+    settings = mkOption {
+      type = settingsFormat.type;
+      default = { };
+      description = ''
+        Configuration for redfish_exporter, see
+        <https://github.com/jenningsloy318/redfish_exporter>
+        for supported values.
+      '';
+    };
+
+    loglevel = mkOption {
+      type = types.enum [
+        "debug"
+        "info"
+        "warn"
+        "error"
+        "fatal"
+      ];
+      default = "info";
+      description = ''
+        Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal].
+      '';
+    };
+    logformat = mkOption {
+      type = types.str;
+      default = "logger:stderr";
+      description = ''
+        Set the log target and format. Example: "logger:syslog?appname=bob&local=7" or "logger:stdout?json=true"
+      '';
+    };
+
+    webConfigFile = mkOption {
+      type = with types; nullOr path;
```

I really don't like this syntax. I won't block on it, but I'd much prefer

```suggestion
      type = types.nullOr types.path;
```